openwisp_controller/config/apps.py

```diff
@@ -133,3 +139,9 @@ def add_ignore_notification_widget(self):
                 'OPENWISP_NOTIFICATIONS_IGNORE_ENABLED_ADMIN',
                 obj_notification_widget,
             )
+
+    def update_vpn_dhparam(self, instance, created, **kwargs):
+        if created:
+            from .tasks import create_vpn_dhparam
+
+            create_vpn_dhparam.delay(instance.pk)
```

Comment:
this must be called when the transaction is committed to the DB or it will be executed before the VPN object is even created and it won't work. Use transaction.on_commit please.

We also need a `TransactionTestCase` with a functional test to ensure this is executed (just mock `create_vpn_dhparam.delay` and ensure is called).